atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java

```diff
@@ -187,6 +187,7 @@ public Builder socketQueryTimeout(int val){
             config.setTimeBetweenEvictionRunsMillis(TimeUnit.MILLISECONDS.convert(1, TimeUnit.MINUTES));
             config.setNumTestsPerEvictionRun(-1); // Test all idle objects for eviction
             config.setJmxNamePrefix(host);
+            config.setMaxWaitMillis(socketTimeoutMillis);
```

Comment:
Should we be calling `clientPool.borrowObject(socketQueryTimeoutMillis)` in `runWithGoodResource`?
Do we need to retry up to some N times/backoff time limit if `borrowObject ` throws `NoSuchElementException`?